File: libNet/src/main/java/com/effective/android/net/okhttp/api/HttpApi.java
package com.effective.android.net.okhttp.api;

import java.util.Map;


/**
 * 封装常规请求
 * Created by yummyLau on 2018/5/16.
 * Email: yummyl.lau@gmail.com
 * blog: yummylau.com
 */
public interface HttpApi {

    String get(String url);

    String post(String url, String json);

    String post(String url, Map<String, String> mapParams);

    String uploadFile(String url, String pathName, String fileName);
}

File: libNet/src/main/java/com/effective/android/net/okhttp/api/HttpApi.kt
package com.effective.android.net.okhttp.api


/**
 * 封装常规请求
 * Created by yummyLau on 2018/5/16.
 * Email: yummyl.lau@gmail.com
 * blog: yummylau.com
 */
interface HttpApi {

    fun get(url: String): String

    fun post(url: String, json: String): String

    fun post(url: String, mapParams: Map<String, String>): String

    fun uploadFile(url: String, pathName: String, fileName: String): String
}
